mod/consensus-types/pkg/types/header.go

```diff
@@ -176,11 +176,19 @@ func (b *BeaconBlockHeader) GetTree() (*fastssz.Node, error) {
 
 // Equals returns true if the Withdrawal is equal to the other.
 func (b *BeaconBlockHeader) Equals(rhs *BeaconBlockHeader) bool {
-	return b.Slot == rhs.Slot &&
-		b.ProposerIndex == rhs.ProposerIndex &&
-		b.ParentBlockRoot == rhs.ParentBlockRoot &&
-		b.StateRoot == rhs.StateRoot &&
-		b.BodyRoot == rhs.BodyRoot
+	switch {
```

Comment:
not strictly related to the fix, but helpful to harden all the custom Equals functions we wrote